GW2EIEvtcParser/EIData/Buffs/BuffSimulators/BuffSimulationItems/BuffSimulationItemBase.cs

```diff
@@ -1,132 +1,130 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using GW2EIEvtcParser.ParsedData;
+﻿using GW2EIEvtcParser.ParsedData;
 
-namespace GW2EIEvtcParser.EIData.BuffSimulators
+namespace GW2EIEvtcParser.EIData.BuffSimulators;
+
+internal class BuffSimulationItemBase : BuffSimulationItem
 {
-    internal class BuffSimulationItemBase : BuffSimulationItem
+    private readonly AgentItem _src;
+    internal readonly BuffStackItem _buffStackItem;
+    internal readonly long _totalDuration;
+
+    protected internal BuffSimulationItemBase(BuffStackItem buffStackItem) : base(buffStackItem.Start, buffStackItem.Start + buffStackItem.Duration)
     {
-        private readonly AgentItem _src;
-        private readonly AgentItem _seedSrc;
-        private readonly bool _isExtension;
-        private long _totalDuration { get; }
+        _buffStackItem = buffStackItem;
```

Comment:
Is it really worth keeping the whole buff stack item alive for the whole duration of the processing of the log just to have a pointer on SeedSrc?

Reply:
There is some argument to be made for keeping it around, because it might reduce the amount of individual small deallocations, but this is a remnant from me trying to just keep the item around and not copy all the fields (see comments below).
I reverted it to store individual fields.